model/waterfall.go

```diff
@@ -78,7 +114,33 @@ func GetActiveWaterfallVersions(ctx context.Context, projectId string, opts Wate
 		match[VersionRevisionOrderNumberKey] = bson.M{"$gt": opts.MinOrder}
 	}
 
-	pipeline := []bson.M{{"$match": match}}
+	pipeline := []bson.M{}
+
+	if len(opts.Variants) > 0 {
+		match[bsonutil.GetDottedKeyName(VersionBuildVariantsKey, VersionBuildStatusDisplayNameKey)] = bson.M{"$exists": true}
+		pipeline = append(pipeline, bson.M{"$match": match})
+		matchCopy := bson.M{}
```

Comment:
I believe we can remove this copying code since `getBuildDisplayNames` does not accept a pointer

Reply:
I thought so too but no dice, I guess the map pointer is what's passed to the function?